### ISSUE TITLE
Using Chat GPT-4 - Location: Canada EST

### ISSUE THREAD

**Issue by edercarlima** [microsoft/sample-app-aoai-chatGPT#125]
Hi

I'm trying to run the scripts to use the Chat GPT-4 that was released for our subscription being the local Canada EST, but we are not able to create it through the execution of the scripts because it presents the following message when creating the OpenAI service: Sku Standard not allowed .
In the documentation it mentions changing the parameters AZURE_FORMRECOGNIZER_SKU_NAME, AZURE_SEARCH_SKU_NAME and AZURE_OPENAI_SKU_NAME to Premium does not allow the creation, also informing that the SKU is not allowed.
The parameters were configured in this way:

azd env set AZURE_FORMRECOGNIZER_SKU_NAME v2 P1V2
azd env set AZURE_SEARCH_SKU_NAME Premium
azd env set AZURE_OPENAI_SKU_NAME v2 P1V2

The Playground allows you to create the OpenAI service for Canada EST.
Could you help me with this question?

**Comment by ealasgarov** [comment]
Hi,
Are you sure you can deploy gpt-4 in Canada? So far only 2 regions are available: East US & France Central. 
https://learn.microsoft.com/en-us/azure/ai-services/openai/concepts/models#gpt-4-models

**Comment by edercarlima** [comment]
Hi @ealasgarov,

Yes, we received an email informing that use in this region of Canada was released.
In case to use GPT-4 in France region, it is necessary:
  1 - adjust some parameter or configuration file or
  2 - at the time of deployment (via script) we must select the region of France.
As I mentioned earlier, through the Playground we were able to create the OpenAI resource in the Canada region.

**Comment by edercarlima** [comment]
Hi @ealasgarov and @pamelafox 

can you help me with this question?

**Comment by pamelafox** [comment]
Hm, based on an issue I saw in another repo (https://github.com/Azure-Samples/azure-search-openai-demo/issues/497), I wonder if you need to change the code that forms the endpoint URL, like in this issue:
https://github.com/microsoft/sample-app-aoai-chatGPT/issues/82

You could try changing this line:

https://github.com/microsoft/sample-app-aoai-chatGPT/blob/f2b256a13c111a2621b15aba7214562d3be27b3a/app.py#L158

**Comment by nocode99** [comment]
We received the invitation to ChatGPT4 and trying to deploy this to useast2 using the azure developer CLI
![image](https://github.com/microsoft/sample-app-aoai-chatGPT/assets/4175239/1d21f8db-2d59-4e0b-9b45-ecb41213753b)

in `infra/core/ai/cognitiveservices.bicep` on line 34 is:

```
25 @batchSize(1)
26 resource deployment 'Microsoft.CognitiveServices/accounts/deployments@2023-05-01' = [for deployment in deployments: {
27   parent: account
28   name: deployment.name
29   properties: {
30     model: deployment.model
31     raiPolicyName: contains(deployment, 'raiPolicyName') ? deployment.raiPolicyName : null
32   }
33   sku: {
34     name: 'Standard'
35     capacity: deployment.capacity
36   }
37 }]
```

Following some azure troubleshooting guides, I ran `az vm list-skus --location eastus2 --size Standard --all --output table` and get a whole list of SKUs with `NotAvailableForSubscription`. I'm poking around my previous eastus deployment and don't know what this is referring to.

**Comment by phfontes** [comment]
Hello,
I'm also having trouble deploying with gpt-4.
I already opened the ticket and received the email with the release of gpt4
However, when running azd up, the error below appears
Is it necessary somewhere to change this gpt-35-turbo 0301?
Note: I don't have much knowledge
![Uploading deploy.png…]()

**Comment by phfontes** [comment]
Updates
I made some changes and now I'm getting this error:

I've tried changing it to 0 and it says it needs to be between 1 and 1000
Going to the azure resource open ai> explore >Quotas> and gpt-4 20/20 appears
InsufficientQuota: The specified capacity '1' of account deployment is bigger than available capacity '0' for UsageName 'Tokens Per Minute (thousands) - GPT-4'.

**Comment by nocode99** [comment]
I was able to figure this out, but I don't have a perfect answer. I synced the latest changes from this repo and modified `./infra/main.biceps` file:

```
    deployments: [
      {
        name: openAIModel
        model: {
          format: 'OpenAI'
          name: openAIModelName
          version: '0613'
        }
        capacity: 9
      }
    ]
```

1. When I deployed OpenAI manually, the version was 0613 in the console and when I tried deploying, the initial capacity of 30 was too high and 9 being the max.
2. did a search and replace for references to `gpt-35-turbo` to `gpt-4`

However, the app was deployed and I could access it, the chat prompt did not work. I tried matching some of the App Service configurations to my previous deployment and restarted and it still didn't work. Eventually it did so I'm not sure if it just took some time for the LLM to load properly.
